src/casscf/casscf.cc

```diff
@@ -355,35 +350,32 @@ void CASSCF::cas_ci() {
         quiet = false;
     }
 
-    if (options_.get_str("CASSCF_CI_SOLVER") == "FCI") {
+    if (options_->get_str("CASSCF_CI_SOLVER") == "FCI") {
         // Used to grab the computed energy and RDMs.
-        if (options_["AVG_STATE"].size() == 0) {
+        if (options_->psi_options()["AVG_STATE"].size() == 0) {
             set_up_fci();
         } else {
             set_up_sa_fci();
         }
-    } else if (options_.get_str("CASSCF_CI_SOLVER") == "CAS") {
+    } else if (options_->get_str("CASSCF_CI_SOLVER") == "CAS") {
         set_up_fcimo();
-    } else if (options_.get_str("CASSCF_CI_SOLVER") == "ACI") {
-        // ints_->retransform_integrals();
-        std::shared_ptr<FCIIntegrals> fci_ints = get_ci_integrals();
-        AdaptiveCI aci(reference_wavefunction_, options_, ints_, mo_space_info_);
-        aci.set_fci_ints(fci_ints);
+    } else if (options_->get_str("CASSCF_CI_SOLVER") == "ACI") {
+        std::shared_ptr<ActiveSpaceIntegrals> fci_ints = get_ci_integrals();
+        AdaptiveCI aci(state_, scf_info_, options_, mo_space_info_, fci_ints);
```

Comment:
Should be able to pass as_ints_ from constructor of CASSCF

Reply:
Here I can store `fci_ints` in the local `as_ints_`, but I am not sure I can get away with calling the function. The integrals are updated at each cycle and as far as I can tell this is the only way it can happen (by calling `get_ci_integrals`).